### DISCUSSION TITLE
Hosting nodejs on vercel

### DISCUSSION THREAD

**Discussion by janusqa** [vercel/community#4203] [Help]
### Summary

I have a nodeJS application I wish try on vercel that requires a database.  I think I should be able to try it with your beta storage feature.
The way I would like to do it like this:

Project
-------
A preview location (myapp.preview.mydomain.com). Connects to preview database
A production location (myapp.mydomain.com). Connects to production database
A dev location on my machine that can connect to preview database to try things out

Project Database
----------------
A database for preview
A database for production

Is this all possible?

1/
I've been reading your docs and the first problem I run into is environment variables
It seems all the environment variables are stored together and I would have to have more variables then I should need
Example I would like to have PREVIEW variables and PRODUCTION variables.
Do i have to duplicate each variable twice and prefix them with PREVIW_ and PRODUCTION_ ?

2/
I have one github repo with both the frontend and the backend in it I just want to connect the backend folder to vercel
Would I have to seperate the repo into two repos to acheive this?
main
 - frontend
 - backend
 
3/
I have a domain (mydomain.com).
I would like to have production at my (myapp.mydomain.com) I can configure any records needed at the registrar
I would like to have preview at (myapp.preview.mydomain.com) I can configure any records needed at the registrar
I see a domains section where I can add domains to an app, so I think this is already possible via the A/CNAME method?
https://vercel.com/docs/projects/domains/add-a-domain

### Steps to Reproduce

n/a

**Comment by amyegan** [comment]
All great questions, @janusqa!

### [Environment Variables](https://vercel.com/docs/projects/environment-variables#)

You would need to have `preview` and `production` copies of the variables to set different values, but you do not need to prefix the variables since they will be automatically applied only in the appropriate environment.

For example, I might have a variable named `TEST_KEY`. I could have a `preview` copy of that variable with the value `preview-key-value` and a `production` copy of the variable with the value `production-key-value`. Then, I could reference the variable `TEST_KEY` in my code and the value `preview-key-value` would come through automatically on `preview` deployments. Once I made a `production` deployment of the same code, the value `production-key-value` would come through instead for that deployment.

### [Monorepos](https://vercel.com/docs/monorepos#)

You can deploy all or part of a monorepo with Vercel. In many cases, it's as easy as setting a Root Directory and allowing the builder to include source files outside of the Root Directory in the Build Step.

### [Domains](https://vercel.com/docs/projects/domains#)

Yes, you can use A/CNAME records to connect custom domain from an external provider to your Vercel project.

Some people like to set up a specific branch for staging previews. You can learn more about how to configure that at https://vercel.com/guides/set-up-a-staging-environment-on-vercel
